src/util/simple_timer.c

```diff
@@ -1,49 +1,60 @@
-/* -------------------------------------------------------------------------- */
-/* GraphBLAS/Demo/simple_timer.c: a timer for performance measurements        */
-/* -------------------------------------------------------------------------- */
+//------------------------------------------------------------------------------
+// GraphBLAS/Demo/Source/simple_timer.c: a timer for performance measurements
+//------------------------------------------------------------------------------
 
-/* SuiteSparse:GraphBLAS, Timothy A. Davis, (c) 2017, All Rights Reserved.    */
-/* http://suitesparse.com   See GraphBLAS/Doc/License.txt for license.        */
+// SuiteSparse:GraphBLAS, Timothy A. Davis, (c) 2017-2021, All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
 
-/* -------------------------------------------------------------------------- */
+//------------------------------------------------------------------------------
 
-/* simple_timer:  a portable timer for accurate performance measurements */
+// simple_timer:  a portable timer for accurate performance measurements
 
 #include "simple_timer.h"
 
-/* -------------------------------------------------------------------------- */
-/* simple_tic: return the current wallclock time in high resolution           */
-/* -------------------------------------------------------------------------- */
+//------------------------------------------------------------------------------
+// simple_tic: return the current wallclock time in high resolution
+//------------------------------------------------------------------------------
 
-void simple_tic         /* returns current time in seconds and nanoseconds */
+void simple_tic         // returns current time in seconds and nanoseconds
 (
-    double tic [2]      /* tic [0]: seconds, tic [1]: nanoseconds */
+    double tic [2]      // tic [0]: seconds, tic [1]: nanoseconds
 )
 {
 
     #if defined ( _OPENMP )
 
-        /* OpenMP is available; use the OpenMP timer function */
+        // OpenMP is available; use the OpenMP timer function
         tic [0] = omp_get_wtime ( ) ;
         tic [1] = 0 ;
 
-    #elif defined ( __linux__ )
+    #elif defined ( __linux__ ) || defined ( __GNU__ )
 
-        /* Linux has a very low resolution clock() function, so use the high
-           resolution clock_gettime instead.  May require -lrt */
+        // Linux has a very low resolution clock() function, so use the high
+        // resolution clock_gettime instead.  May require -lrt
         struct timespec t ;
         clock_gettime (CLOCK_MONOTONIC, &t) ;
         tic [0] = (double) t.tv_sec ;
         tic [1] = (double) t.tv_nsec ;
 
+    #elif defined ( __MACH__ ) && defined ( __APPLE__ )
```

Comment:
Is this also faster on macOS than using a C11 clock? In the comment below it is stated that the C11 clock on macOS also uses some high-resolution clock.

Reply:
this fixes a bug reporting incorrect timing on MacOS

Reply:
Thanks for the explanation!